### DISCUSSION TITLE
Mention pitfalls of subclassing group

### DISCUSSION THREAD

**Discussion by PythonCoderAS** [Rapptz/discord.py#9383] [Ideas]
I decided to subclass `discord.app_commands.Group`. I tried making some instance attributes tied to an instance of the group, but ran into major pitfalls that eventually I solved by just making class variables instead of instance variables. Could this pitfall be noted in the docs?

**Comment by Rapptz** [comment]
Considering I have no idea what you're talking about, not really. I suggest filing a bug report if something does not work as expected or to get more clarification. As it stands right now this is too vague to be actionable.

**Reply by PythonCoderAS** [reply]
Here is an example: 

```py
from discord.app_commands import Group, command
from discord.ext.commands import GroupCog

class MyGroup(name="group"):
    def __init__(self):
        self.queue = []
    @command(name="command")
    async def my_command(self, interaction, val: str):
        if val in self.queue:
            return await interaction.response.send_message("Seen value before.")
        else:
            return await interaction.response.send_message("Not seen value before.")

class Cog(GroupCog, group_name="parent_group"):
    group = MyGroup()
```

If you were to call `/parent_group group command`, it will complain that there is no attribute "queue" in self.

**Reply by Rapptz** [reply]
That code should be fixed now. Note that the code as given was invalid but required some fixes to make it work.

**Reply by PythonCoderAS** [reply]
Why is the "Group" subclassing removed?

**Reply by Rapptz** [reply]
I'm not sure I understand.
